demos/rain_prediction.ipynb

```diff
@@ -0,0 +1,467 @@
+   "source": [
+    "!curl -o './data/weather-2010s.json' 'https://archive-api.open-meteo.com/v1/archive?latitude=52.52&longitude=13.41&start_date=2010-01-01&end_date=2019-12-31&hourly=temperature_2m,relativehumidity_2m,rain,pressure_msl,surface_pressure,cloudcover,cloudcover_low,cloudcover_mid,cloudcover_high,windspeed_10m,windspeed_100m,winddirection_10m,winddirection_100m'"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The data is in JSON format so we need to wrangle the data a bit to make it usable. But first we will save it into our lakeFS instance. We will create a new branch, `transform-raw-data`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from lakefs_spec import LakeFSFileSystem\n",
+    "\n",
+    "NEW_BRANCH_NAME = 'transform-raw-data'\n",
+    "\n",
+    "\n",
+    "fs = LakeFSFileSystem()\n",
+    "fs.put('./data/weather-2010s.json',  f'{REPO_NAME}/{NEW_BRANCH_NAME}/weather-2010.json')\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now, on LakeFS in your browser, can change the branch to `transform-raw-data` and see the saved file. However, the change is not yet committed. \n",
+    "While you can do that manually via the uncommitted changes tab in Lakefs, we will take another route to commit. \n",
+    "\n",
+    "We can also do commit changes programmatically. To achieve this, we register a hook. The hook needs to have the signature `(client, context) -> None`. Where the `client` is the respective LakeFS client. The context object contains information about the corresponding resource. \n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from lakefs_client.client import LakeFSClient\n",
+    "from lakefs_spec.client_helpers import commit\n",
+    "from lakefs_spec.hooks import FSEvent, HookContext\n",
+    "\n",
+    "#Define the commit hook\n",
+    "def commit_on_put(client: LakeFSClient, ctx:HookContext) -> None:\n",
+    "    commit_message = f\"Add file {ctx.resource}\"\n",
+    "    print(f\"Attempting Commit: {commit_message}\")\n",
+    "    commit(client, repository=ctx.repository, branch = ctx.ref, message=commit_message)\n",
+    "    \n",
+    "\n",
+    "#Register the commit hook to be executed after a PUT_FILE event\n",
+    "fs.register_hook(FSEvent.PUT_FILE, commit_on_put)\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "When you execute the next cell however, you will see a message indicating that the upload of the resource has been skipped because the file is uploaded to LakeFS already (checksums match). This is useful when we work with large files to reduce the amount of network traffic. Nonetheless, in this specific situation the `PUT` is not executed and neither is our commit hook. \n"
```

Comment:
Whoops, this kind of sucks, right? Defeats the concept of a hook in the first place. Could be worth an issue?

```suggestion
    "However, when you execute the next cell, you will see a message indicating that the upload of the resource has been skipped because the file is uploaded to lakeFS already, and the checksums match. This is useful when we work with large files to reduce the amount of network traffic. Nonetheless, in this specific situation the `PUT` is not executed and neither is our commit hook. \n"
```

Reply:
Opened here https://github.com/appliedAI-Initiative/lakefs-spec/issues/92